slapo/schedule.py

```diff
@@ -1331,12 +1331,15 @@ def _consolidate_and_broadcast(sch: Schedule):
             orig_shape = (
                 param.orig_shape if hasattr(param, "orig_shape") else param.shape
             )
+            new_param = nn.Parameter(
+                torch.empty(orig_shape, dtype=param.dtype, device=local_rank)
+            )
             sch.mod.register_parameter(
                 param_name,
-                nn.Parameter(
-                    torch.empty(orig_shape, dtype=param.dtype, device=local_rank)
-                ),
+                new_param,
             )
+            if hasattr(param, "replicated_param") and param.replicated_param:
+                new_param.replicated_param = True
```

Comment:
This is a bit hacky. We should add a TODO or FIXME saying that we will have a mechanism to maintain a list of user-specified tags.